sunpy/net/dataretriever/sources/kanzelhohe.py

```diff
@@ -0,0 +1,125 @@
+#This module was developed with funding provided by
+#the Google Summer of Code 2016.
+
+__author__ = "Sudarshan Konge"
+__email__ = "sudk1896@gmail.com"
+
+import datetime
+import urllib2
+
+
+from sunpy.net.dataretriever.client import GenericClient
+from sunpy.util.scraper import Scraper
+
+from sunpy.time import TimeRange
+
+__all__ = ['KanzelhoheClient']
+
+class KanzelhoheClient(GenericClient):
+    """
+    Returns a list of URLS to Kanzelhohe H-alpha files corresponding to value of input timerange.
+    URL source: `http://cesar.kso.ac.at/`.
+
+    The earliest data for H-alpha 2k - 20-Jul-2000
+                          Ca-II k - 31-Jul-2010
+                          Continuum - 7-Jan-2011
+
+    Parameters
+    ----------
+    timerange: sunpy.time.TimeRange
+        time range for which data is to be downloaded.
+        Example value - TimeRange('2015-12-30 00:00:00','2015-12-31 00:01:00')
+    
+    Instrument: Fixed argument = 'kanzelhohe'
+
+    Wavelength: Fixed argument = astropy.units.quantity.Quantity
+                The physical value of wavelength will belong to any of [5460, 6563, 32768]
+                and units will be Angstroms.
+
+    Returns
+    -------
+    urls: list
+    list of urls corresponding to requested time range.
+
+    Examples
+    --------
+    >>> from sunpy.net import Fido
+    >>> from sunpy.net import attrs as a
+
+    >>> results = Fido.search(a.Time('2015/12/28 00:00:00', '2015/12/28 00:03:00'), a.Instrument('kanzelhohe'), a.Wavelength(6563*u.AA))
+    >>> print(results)
+    >>> [<Table length=1>
+        Start Time           End Time              Source        Instrument
+        str19               str19                str21           str10   
+    ------------------- ------------------- --------------------- ----------
+    2015-12-28 00:00:00 2015-12-29 00:00:00 Global Halpha Network Kanzelhohe]
+    
+    >>> response = Fido.fetch(results)
+    """
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        returns list of urls corresponding to given TimeRange.
+        """
+        wave = int(kwargs['Wavelength'].min.value)
+        table = {6563:'halpha2k/recent', 32768:'caiia', 5460:'phokada'}
+        table1 = {6563:'halph_fr', 32768:'caiik_fi', 5460:'bband_fi'}
```

Comment:
Okay, The tables are basically for building the ```url_pattern``` i.e. regex.
For H-alpha 2k in different parts of the regex(in prefix and suffix) you can replace the wavelength with the value that it maps to in the given dictionary.
You do the same for ca-II and continuum, If you study the url_patterns carefully, you would understand. Let me know, if I can explain it more clearly.